tools/tests/com.liferay.ide.project.core.tests/src/com/liferay/ide/project/core/tests/modules/jsf/NewLiferayJSFModuleProjectOpTests.java

```diff
@@ -0,0 +1,209 @@
+import static org.junit.Assert.assertTrue;
+
+import com.liferay.ide.core.ILiferayProjectImporter;
+import com.liferay.ide.core.LiferayCore;
+import com.liferay.ide.core.util.CoreUtil;
+import com.liferay.ide.core.util.ZipUtil;
+import com.liferay.ide.project.core.modules.jsf.NewLiferayJSFModuleProjectOp;
+import com.liferay.ide.project.core.modules.jsf.NewLiferayJSFModuleProjectOpMethods;
+
+import java.io.File;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Set;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.FileLocator;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.sapphire.DefaultValueService;
+import org.eclipse.sapphire.PossibleValuesService;
+import org.eclipse.sapphire.modeling.Status;
+import org.eclipse.sapphire.platform.ProgressMonitorBridge;
+import org.junit.Test;
+
+/**
+ * @author Simon Jiang
+ */
+public class NewLiferayJSFModuleProjectOpTests
+{
+
+    @Test
+    public void testNewLiferayJSFModuleProjectOpProjectName() throws Exception
+    {
+        NewLiferayJSFModuleProjectOp op = NewLiferayJSFModuleProjectOp.TYPE.instantiate();
+        op.setProjectName( "Test1" );
+        Status projectNameOkValidationStatus1 = op.getProjectName().validation();
+        assertEquals( projectNameOkValidationStatus1.message(), "ok" );
+
+        op.setProjectName( "#Test1" );
+        Status projectNameErrorValidationStatus = op.getProjectName().validation();
+        assertEquals( projectNameErrorValidationStatus.message(), "The project name is invalid." );
+
+        op.setProjectName( "Test1_Abc" );
+        Status projectNameOkValidationStatus2 = op.getProjectName().validation();
+        assertEquals( projectNameOkValidationStatus2.message(), "ok" );
+    }
+
+    @Test
+    public void testNewLiferayJSFModuleProjectOpProjectExisted() throws Exception
+    {
+        NewLiferayJSFModuleProjectOp existedProjectop = NewLiferayJSFModuleProjectOp.TYPE.instantiate();
+        existedProjectop.setProjectName( "Test2" );
+        Status exStatus = NewLiferayJSFModuleProjectOpMethods.execute(
+            existedProjectop, ProgressMonitorBridge.create( new NullProgressMonitor() ) );
+        assertTrue( exStatus.ok() );
+
+        IProject existedProject = CoreUtil.getProject( existedProjectop.getProjectName().content() );
+        assertNotNull( existedProject );
+
+        NewLiferayJSFModuleProjectOp newProjectNameop = NewLiferayJSFModuleProjectOp.TYPE.instantiate();
+        newProjectNameop.setProjectName( "Test2" );
+        Status projectNameExistedValidationStatus = newProjectNameop.getProjectName().validation();
+        assertEquals(
+            projectNameExistedValidationStatus.message(), "A project with that name(ignore case) already exists." );
+    }
+
+    @Test
+    public void testNewLiferayJSFModuleProjectOpBuildType() throws Exception
```

Comment:
No need to test this, we can just have some test that test gradle-jsf-module and some that create maven-jsf-module and that will be enough to make sure that both buildTypes are checked.